forgettable-frontend/src/pages/CreateEncounterPage/CreateEncounterPage.js

```diff
@@ -0,0 +1,204 @@
+/* eslint-disable max-len */
+import React, {useEffect} from 'react';
+import classes from './CreateEncounterPage.module.css';
+import {Card} from '@mui/material';
+import TextField from '@mui/material/TextField';
+import CustomButton from '../../components/CustomButton/CustomButton';
+import {Autocomplete} from '@mui/material';
+import {createEncounter, getAllPersons} from '../../services';
+import {Link} from 'react-router-dom';
+
+export default function CreateEncountersPage() {
+  const [optionsList, setOptionsList] = React.useState([]);
+  const [encounter, setEncounter] = React.useState({
+    title: '',
+    date: null,
+    location: null,
+    description: '',
+    persons: [],
+  });
+  const [isSubmittable, setIsSubmittable] = React.useState(false);
+  const [showWarning, setShowWarning] = React.useState(false);
+
+  async function getData() {
+    let peopleResult = [];
+    peopleResult = await getAllPersons();
+
+    // format the people for the autocomplete list
+    const options = [];
+    peopleResult.forEach((person) => {
+      options.push({label: person.first_name, id: person._id});
+    });
+
+    setOptionsList(options);
+  }
+
+  useEffect(() => {
+    try {
+      getData();
+    } catch (err) {
+      console.log(err);
+    }
+  }, []);
+
+  const handlePersonChange=(event, value)=>{
+    if (encounter.title != '' && value.length != 0) {
+      setIsSubmittable(true);
+      setShowWarning(false);
+    }
+    const personArr = [];
+    value.forEach((person)=>{
+      personArr.push(person.id);
+    });
+    setEncounter({...encounter, persons: personArr});
+  };
+
+  const handleDateChange=(event)=>{
+    setEncounter({...encounter, date: event.target.value});
+  };
+
+  const handleTitleChange=(event)=>{
+    if (event.target.value != '' && encounter.persons.length != 0) {
+      setIsSubmittable(true);
+      setShowWarning(false);
+    }
+    setEncounter({...encounter, title: event.target.value});
+  };
+
+  const handleLocationChange=(event)=>{
+    setEncounter({...encounter, location: event.target.value});
+  };
+  const handleDescriptionChange=(event)=>{
+    setEncounter({...encounter, description: event.target.value});
+  };
+
+  const handleSaveClick = (event) => {
+    if (!encounter.title || encounter.persons.length === 0) {
```

Comment:
Non blocking suggestion: 
```suggestion
    if (!encounter.title || encounter.persons?.length === 0) {
```
So the app doesn't crash when encounter doesn't have a persons property (unlikely to happen, we try to prevent this)

Reply:
I'm the one setting encounter here :) so persons always exists, i initialise it as an empty array

Reply:
Ahh nice, in that case no worries :)